internal/resources/otelcollector/core/core.go

```diff
@@ -0,0 +1,168 @@
+package core
+
+import (
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+
+	collectorconfig "github.com/kyma-project/telemetry-manager/internal/otelcollector/config"
+)
+
+const (
+	configMapKey            = "relay.conf"
+	configHashAnnotationKey = "checksum/config"
+	collectorUser           = 10001
+	collectorContainerName  = "collector"
+)
+
+func MakeDefaultLabels(baseName string) map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/name": baseName,
+	}
+}
+
+func MakePodSpec(baseName, image, priorityClassName string, resources corev1.ResourceRequirements) corev1.PodSpec {
+	labels := MakeDefaultLabels(baseName)
+	affinity := makePodAffinity(labels)
```

Comment:
Not required for the DaemonSet. But probably would also not cause any harm to keep it. Please check again.